### PR TITLE
Billing History: Add display_brand to individual billing receipts

### PR DESCRIPTION
**Pull request by jjchrisdiehl**
> [!warning]
> This PR relies on the data provided by D160555-code, do not merge until **_after_** the diff is deployed.

Follow up to the Stripe Cobadge rollout - this PR will update the card brand displayed on individual receipts under `/purchass/billing` to show the display_brand value if available.

**Note**: still a WIP, I noticed this isn't working for site specific billing history. Will update soon.

Related to https://github.com/Automattic/payments-shilling/issues/3091

## Proposed Changes

* The diff above adds a `cc_display_brand` value to the `billing-history` endpoint, which is then saved in our redux state under `billingTransactions`
* The `transaction` prop is filtered and paginated, then passed to the `<ReceiptBody>` where it is used to create the `MASTERCARD ending in 1001` payment method for the individual billing receipts
* When the `ReceiptPaymentMethod` component is being generated, this change first checks if `cc_display_brand` exists and is not null and uses it before falling back to the `cc_type`
* Also, this change replaces any underscores with a space, then converts the string to uppercase

## Testing Instructions
* Apply this diff to your sandbox D160555-code
* Complete a test purchase with a Cartes Bancaires payment method (see below for instructions)
* Go to /me/purchases/billing and click on `View Receipt` for the recent purchase
* Ensure that the `CARDBRAND ending in 1001` shows as `CARTES BANCAIRES ending in 1001`
* Do the same for the site level billing history at `/purchases/billing-history/[site-name]`

### Test card instructions
- Proxy from the cdg server
- Go to checkout, type in the following test card number 4000002500001001
- Click on network selector next to the card number and select Cartes Bancaires
- Complete checkout to save card

### CODE REVIEW

**Review by michaeldcain — APPROVED** [review]
This LGTM after D160555-code merges… unless you decide to handle the formatting on the backend.

**Comment by jjchrisdiehl** [comment]
> This LGTM after [D160555-code](https://href.li/?https://code.a8c.com/D160555) merges… unless you decide to handle the formatting on the backend.

That's a good question 🤔 I'm more inclined to keep whatever string Stripe passes to us 'as-is' and then modify it for display on the front end. But I'm not sure what the best practice here would be.

**Pull request merged**
